### ISSUE TITLE
Connection with my own repo fails

### ISSUE THREAD

**Issue by MarcelStructr** [aegif/CmisSync#644]
Hello! 

I am using Version 2.6.5.0. 

Tried to connect my repo into CmisSync, but this error occures:

---
Value cannot be null.
Parameter name: stream
Sorry, CmisSync can not find a CMIS server at this address.
Please ask your server administrator for the CMIS endpoint URL.
If you are sure about the address, open it in a browser and post
the resulting XML on the cmissync forum.
---

I used the Login-URL for Browser and AtomPub Binding, which also works flawless for CMIS Workbench. Wanted to test it also for CmisSync, but doesn't work yet.

Any suggestions?

**Comment by nicolas-raoul** [comment]
Hello Marcello,
What repository product are you using?
Cheers!

**Comment by MarcelStructr** [comment]
Hello Nicolas, 

I am using "Structr".
I am currently working on integrating the CMIS Server into the product. https://github.com/MarcelStructr/structr

I can post the resulting XML after putting the Login URL into the browser, if you want.

Thanks for your help!

**Comment by nicolas-raoul** [comment]
In CMIS Workbench, please press "TCK" and send me the resulting HTML,
thanks!

**Comment by MarcelStructr** [comment]
http://www.file-upload.net/download-11091827/cmistck6656285434632739829.html.html

Here the requested HTML-File.

**Comment by nicolas-raoul** [comment]
As you can see by opening this HTML file, your server still fails some unit
tests.
Fixing them should make your server work with CmisSync, in particular
please focus on the "Root Folder Test" and "CRUD Test Group".
CmisSync is not very demanding, so it might work even when not *everything*
is green.
Keep up the great work, cheers!

**Comment by nicolas-raoul** [comment]
Also, what exact URL did you enter in the "Add a remote folder" wizard?

**Comment by MarcelStructr** [comment]
Good Idea to change the title of the Issue this way! "My own repo" is not actually true, it is a repo with a name and of a whole company behind! 

I know, that some unit tests still fail. I am wondering why on the CMIS Workbench it already works, though. Anyway i will follow your advice and fix the last failing Unit Tests. 

I used the URL: http://localhost:8082/structr/cmis/atom or http://localhost:8082/structr/cmis/browser

Thank you for your cooperation and have a nice day.

**Comment by nicolas-raoul** [comment]
CMIS Workbench only uses a very small subset of operations. CmisSync uses a few more.

Also, could you please use Wireshark to capture CMIS requests/responses, and if you think CmisSync's request is malformed (not OK in regard to the CMIS standard) please post it to Stack Overflow asking if it is indeed malformed or not. Thanks!

**Comment by MarcelStructr** [comment]
I am pretty sure, that CmisSync's request works properly. 

I already tested CmisSync with following CMIS Server, which also uses the OpenCMIS Server Framework: https://github.com/cmisdocs/ServerDevelopmentGuide
It calls the credentials username and password with the exact same way, like in my code, but it works! So I am pretty sure the problem doesn't lie on the Client side.

Anyway I will check it out, after fixing the unit test fails and I would be glad, to help you out.

**Comment by MarcelStructr** [comment]
Hello Nicolas, 

meanwhile I fixed all TCK failures caused with Structr as CMIS Server. And the Login works! I get access to the files.

BUT: 
The issue with not getting the login credentials is still present (at the moment a user can only login as a public user with minor allowable actions).

Just for refreshing: 

 ```
getService(CallContext cc) {
cc.getUserName(); -> is always null, doesn't matter what i type into "User" and "Password" at CmisSync
cc.getPassword(); -> same
}
```

At this point no username and password is delivered. It works with CMIS Workbench though. 

---------

I also used Wireshark and inspected a little the CMIS packages. I found following: 

http://www2.pic-upload.de/img/29151909/cmisworkbench_package.png
This is a package received from the CMIS Workbench, where you can see the login credentials admin:admin. 

http://www2.pic-upload.de/img/29151911/cmissync_package.png
This is a package received from CmisSync, but there are no such credentials (at least I couldn't find the information).


Do you have any idea? 

Thanks for reading and best regards

**Comment by nicolas-raoul** [comment]
For both, could you please right-click, choose "Follow TCP stream", and
post the whole results as text?

**Comment by MarcelStructr** [comment]
Here we go.
[workbench.txt](https://github.com/aegif/CmisSync/files/67903/workbench.txt)
[cmissync.txt](https://github.com/aegif/CmisSync/files/67904/cmissync.txt)

I will be on vacation over the christmas days, so I will probably not reply immediately.

Thank you anyway for your help. Merry christmas and a happy new year! 
Marcel

**Comment by MarcelStructr** [comment]
Hello Nicolas, just wanted to wish you a happy new year and tell you, that I am back from vacation.

**Comment by nicolas-raoul** [comment]
Happy New Year Marcello! :-)
